model/sshable.rb

```diff
@@ -138,6 +138,13 @@ def invalidate_cache_entry
     Thread.current[:clover_ssh_cache]&.delete([host, unix_user])
   end
 
+  def available?
```

Comment:
This feels like a weird way to check availability. It seems we use this to ensure the VM is up before running actual command through SSH. In that case, couldn't we just rescue the error from that command (which is what we do usually in similar scenarios)?

Since this falls into programming model bucket, I wonder what @fdr would suggest for this case.

Reply:
Yeah, I was surprised with the lack of a general purpose `available?` method for sshable. In this change, since I do a reboot and wait for it to finish, I would need something like this anyway, so putting into the sshable felt better.

Reply:
Not opposed in principle, but looking at how it is used here, I can see why it was not written earlier: why start by running the script you want rather than saying: "can I run my command?" first, then running it anyway? You always have to deal with a possible error for each cmd, why add another one? This reminds me of the unix classic, people that check: "stat" on a file before "unlink" (that is, `if [-f file] rm file`), only to then be subject to race conditions and required to handle exceptions from "unlink" anyway. The most common practice is to go ahead and try to do what you need to do and handle the exceptions once.

In theory I can imagine `available?` being useful if you have a number of complicated side effects outside/before the ssh, and you want to heuristically check whether you have a good chance at succeeding with the ssh command before going ahead. But in practice, given that every ssh command is liable to fail and needs idempotency handling, this seems to never quite happen.

Reply:
In this case, we know the ssh command will fail, until the reboot finishes. There are 2 cases for it: in the host_nexus reboot and here. I think explicitly checking the availability in these cases makes the code a little bit more clear. 

about logistics: let me close this conversation and continue the discussion here: https://github.com/ubicloud/ubicloud/pull/2243#discussion_r1869512597